source/server-selection/server-selection.rst

```diff
@@ -750,17 +758,20 @@ as follows:
 
 3. Find suitable servers by topology type and operation type
```

Comment:
This is going to include tag selection, etc.  Do you want that to happen after the user-provided function?

Reply:
The intention is to first apply the read preference, including tag sets, then the application-provided selector.  

This feature is mostly unnecessary for replica sets because the same goals can be achieved by applying tag sets.  Since mongos has no tag set support, the feature lets the application build it themselves for mongos selection via application-defined logic.  So whether it comes before or after tag sets is perhaps not that important as it's likely to be used mostly for mongos selection.

For those using this feature with replica sets, one disadvantage of moving this selector before the read preference selection is that the it may be considering servers that will ultimately be filtered out by the read preference selection, leaving nothing left to select from.

Reply:
@xdg are you still requesting a change here or are you convinced that it can remain as is?

Reply:
I wasn't requesting a change, just asking you to confirm the intention.

Reply:
Your review is marked as "Requested changes", hence my question. :)